### DISCUSSION TITLE
Editing a depth-image to remove some elements

### DISCUSSION THREAD

**Discussion by Husky110** [Mikubill/sd-webui-controlnet#2021] [Q&A] [answered]
Hi there,  

I am new to the whole stable diffusion-thing and stumbled across ControllNet and Regional Prompter. So if I get this right, there seems to be (at least at the moment) no way to tell ControllNET something like "Hey, please use only THAT SPECIFIC part of your preprocessor-image on the image-generation.", but I was wondering if there might be another way to acchieve this. So my question is that if and how it is possible to edit an exisiting depthmap-image to make parts of it appear to SD as "creative-areas" where only the prompt is applied. I tried to do that by blackening out a part of the depthmap-image, but that resulted in a cursed image, where the blackend part was created as a rectangluar thing... nightmare-stuff... :)

**Comment by redhermes** [comment]
I did the exact same thing with the same result :-)  I also tried blurring the edges but that didn't' work either.

**Reply by Husky110** [reply]
see the marked answer. might help you aswell. :)

**Comment by Husky110** [comment]
@redhermes - Okay, that only confirms to me that it seems like those are not meant to be edited, at least not in an obvious way...  
I still wonder why it seems to be so hard applying those ControllNets only partially. TBH I have no idea of the exact math that is going on in the background, but can't those ControllNets be treated like a LoRa? Cause then it would allow for the partial ControllNet everyone is after... (when used in combination with Regional Prompter).
But maybe there is a way to edit those depth-masks...? :)

**Comment by Husky110** [comment, marked as answer]
Okay - I think I found a way to acchieve what I wanted. This might not work out in every situation, but the general concept works. The first thing one has to do is to NOT use automatic1111-webui for the actual imagecreation, but comfyui. Once you have a controlnet-depth-image, take that image and crop out the part you want to have. Then resize that part to 512x512 (or an aspect-ratio where 512 is one of the sides - 1024 for xl-models) and let comfyui build that part, followed by an latentimageupscaling to the ratio of the image you actually want. It's basically the same concept as "area composition" which is explained here: https://comfyanonymous.github.io/ComfyUI_examples/area_composition/

**Reply by Husky110** [reply]
A Youtube-Video of the concept can be found here aswell: https://www.youtube.com/watch?v=1SwDCqgXZ-M
